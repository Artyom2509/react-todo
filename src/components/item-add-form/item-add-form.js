import React, { Component } from 'react';
import './item-add-form.css';

export default class ItemAddForm extends Component {
	constructor(props) {
		super(props);

		this.state = {
			label: '',
		};
	}

	onLabelChange = (e) => {
		this.setState({
			label: e.target.value,
		});
	};

	onSubmit = (e) => {
		e.preventDefault();

		if (this.state.label.trim()) {
			this.props.onItemAdd(this.state.label.trim());
			this.setState({
				label: '',
			});
		}
	};

	render() {
		return (
			<form className="item-add-form d-flex" onSubmit={this.onSubmit}>
				<input
					type="text"
					className="form-control"
					onChange={this.onLabelChange}
					placeholder="Whats needs to be done"
					value={this.state.label}
				/>
				<button className="btn btn-outline-secondary">Add Item</button>
			</form>
		);
	}
}
