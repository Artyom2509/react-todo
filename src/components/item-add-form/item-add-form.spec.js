import React from 'react';
import { render, shallow } from 'enzyme';
import { ItemAddForm } from './item-add-form';

it('snapshot', () => {
	const wrapper = render(<ItemAddForm />);
	expect(wrapper).toMatchSnapshot();
});

describe('Testing events', () => {
	it('should change state on change event', () => {
		const wrapper = shallow(<ItemAddForm />);
		const input = wrapper.find('.item-add-form input');
		input.simulate('change', { target: { value: 'work' } });
		expect(wrapper.state().label).toEqual('work');
	});
	it('should call function on submit', () => {
		const addItem = jest.fn();
		const wrapper = shallow(<ItemAddForm store={{ addItem }} />);
		const input = wrapper.find('.item-add-form input');
		input.simulate('change', { target: { value: 'work' } });
		wrapper.simulate('submit', { preventDefault: () => {} });
		expect(addItem).toBeCalled();
	});
});
