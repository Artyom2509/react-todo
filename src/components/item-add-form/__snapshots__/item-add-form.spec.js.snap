// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "item-add-form d-flex",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "form-control",
          "placeholder": "Whats needs to be done",
          "type": "text",
          "value": "",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "btn btn-outline-secondary",
          },
          "children": Array [
            Node {
              "data": "Add Item",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "placeholder": undefined,
          "type": undefined,
          "value": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "btn btn-outline-secondary",
        },
        "children": Array [
          Node {
            "data": "Add Item",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "form-control",
            "placeholder": "Whats needs to be done",
            "type": "text",
            "value": "",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
            "value": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "placeholder": undefined,
            "type": undefined,
            "value": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
